src/chainstate/stacks/db/accounts.rs

```diff
@@ -0,0 +1,829 @@
+            microblock_fees_produced = fees_produced.checked_mul(2).expect("FATAL: failed to calculate microblock STX stream fee") / 5;
+        }
+
+        let miner_reward = MinerReward {
+            address: miner.address.clone(),
+            coinbase: coinbase_reward,
+            tx_fees_anchored_shared: anchored_tx_fee_shared,
+            tx_fees_anchored_exclusive: anchored_tx_fee_exclusive,
+            tx_fees_streamed_produced: microblock_fees_produced,
+            tx_fees_streamed_confirmed: microblock_fees_confirmed,
+            vtxindex: miner.vtxindex
+        };
+        miner_reward
+    }
+
+    /// Find the latest miner reward to mature, assuming that there are mature rewards.
+    /// Returns a list of payments to make to each address -- miners and user-support burners
+    pub fn find_mature_miner_rewards<'a>(tx: &mut StacksDBTx<'a>, tip: &StacksHeaderInfo) -> Result<Option<Vec<MinerReward>>, Error> {
+        if tip.block_height < MINER_REWARD_MATURITY + MINER_REWARD_WINDOW + 1 {
+            // no mature rewards exist
+            return Ok(None);
+        }
+
+        let matured_miners = StacksChainState::get_scheduled_block_rewards_in_fork(tx, tip, tip.block_height - MINER_REWARD_MATURITY - MINER_REWARD_WINDOW)?;
+        if matured_miners.len() == 0 {
+            // no sortition happened here
+            return Ok(None);
+        }
+
+        let mut scheduled_payments = vec![];
+        for i in 0..MINER_REWARD_WINDOW {
+            let height = tip.block_height - MINER_REWARD_MATURITY - MINER_REWARD_WINDOW + i;      // safe due to the above check
+            let scheduled_rewards = 
+                if i == 0 {
+                    matured_miners.clone()
+                }
+                else {
+                    StacksChainState::get_scheduled_block_rewards_in_fork(tx, tip, height)?
+                };
+
+            assert!(scheduled_rewards.len() > 0);
+
+            let mut miner_reward_opt = None;
+            for reward in scheduled_rewards.iter() {
+                if reward.miner {
+                    miner_reward_opt = Some(reward.clone());
+                    break;
+                }
+            }
+            
+            let miner_reward = miner_reward_opt.expect(&format!("FATAL: missing miner reward for block {}", height));
+            let mut user_burns = vec![];
+            for reward in scheduled_rewards.iter() {
+                if !reward.miner {
+                    user_burns.push(reward.clone());
+                }
+            }
+
+            assert_eq!(user_burns.len(), scheduled_rewards.len() - 1);
+            scheduled_payments.push((miner_reward, user_burns));
+        }
+
+        let mut rewards = vec![];
+        for matured_miner in matured_miners {
+            let reward = StacksChainState::calculate_miner_reward(&matured_miner, &scheduled_payments);
+            rewards.push(reward);
+        }
+        Ok(Some(rewards))
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
These tests need to be correct to compensate for the broken `calculate_miner_reward` method.